extensions/mongodb/src/test/java/com/hazelcast/jet/mongodb/dataconnection/MongoDataConnectionSslTestBase.java

```diff
@@ -0,0 +1,145 @@
+ * limitations under the License.
+ */
+package com.hazelcast.jet.mongodb.dataconnection;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.hazelcast.config.DataConnectionConfig;
+import com.hazelcast.jet.SimpleTestInClusterSupport;
+import com.hazelcast.jet.test.IgnoreInJenkinsOnWindows;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+
+import javax.annotation.Nonnull;
+import javax.net.ssl.SSLContext;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+
+import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
+import static com.hazelcast.jet.mongodb.AbstractMongoTest.TEST_MONGO_VERSION;
+import static com.hazelcast.jet.mongodb.impl.Mappers.defaultCodecRegistry;
+import static com.hazelcast.test.DockerTestUtil.assumeDockerEnabled;
+import static org.testcontainers.containers.BindMode.READ_WRITE;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class, IgnoreInJenkinsOnWindows.class})
+public abstract class MongoDataConnectionSslTestBase extends SimpleTestInClusterSupport {
+
+    protected static final String DATABASE = "MongoDataConnectionSslTest";
+    protected static final String TEST_COLLECTION = "testCollection";
+    protected static final Logger LOGGER = LoggerFactory.getLogger(MongoDataConnectionSslTestBase.class);
+    protected static String connectionString;
+    protected static String trustStoreLocation;
+    protected static String keyStoreLocation;
+
+    @BeforeClass
+    public static void setUp() {
+        try (
+                InputStream resourceTS = MongoDataConnectionSslTestBase.class.getResourceAsStream("/certs/ca.p12");
+                InputStream resourceKS = MongoDataConnectionSslTestBase.class.getResourceAsStream("/certs/localhost.p12")
+        ) {
+            File tempFileTS = File.createTempFile("MongoDataConnectionSslTest", "jks");
+            Files.copy(resourceTS, tempFileTS.toPath(), StandardCopyOption.REPLACE_EXISTING);
+            File tempFileKS = File.createTempFile("MongoDataConnectionSslTest", "jks");
+            Files.copy(resourceKS, tempFileKS.toPath(), StandardCopyOption.REPLACE_EXISTING);
+            trustStoreLocation = tempFileTS.toString();
+            keyStoreLocation = tempFileKS.toString();
```

Comment:
Why do you copy the files to temp first? Why not point directly to target/test-classes/ or call `Class#getResource` and use the URL it returns?

Reply:
https://github.com/hazelcast/hazelcast/pull/25301/commits/e74b2d6a8f2a0867acbb4fe89a6ab8e96fae08fe